src/richie/apps/search/filter_definitions/base.py

```diff
@@ -0,0 +1,382 @@
+            },
+            self.filter_definitions.values(),
+            {},
+        )
+
+    def get_query_fragment(self, data):
+        """
+        Aggregate query fragments from the nested children in a list of mappings.
+
+        Returns:
+            List[Dict]: a list of one dictionary mapping the name of the nesting wrapper with a
+                nested query cumulating all the active filters among the nested children.
+
+        For example, if the children return query fragments as follows:
+            - availability:
+                {
+                    'key': 'availability',
+                    'fragment': [
+                        {
+                            'bool': {
+                                'must': [{'range': {'course_runs.end': {'lte': '2019-03-08}}}]
+                            }
+                        }
+                    ]
+                }
+            - languages:
+                {
+                    'key': 'languages',
+                    'fragment': [
+                        {
+                            'bool': {
+                                'must': [{'terms': {'course_runs.languages': ['en', 'fr']}}]
+                            }
+                        }
+                    ]
+                }
+
+        Then the nesting wrapper will return the following list with one aggregated nested query:
+        [
+            {
+                'key': 'course_runs',
+                'fragment': [
+                    {
+                        'nested': {
+                            'path': 'course_runs',
+                            'query': {
+                                'bool': {
+                                    'must': [
+                                        {'range': {'course_runs.end': {'lte': '2019-03-08}}},
+                                        {'terms': {'course_runs.languages': ['en', 'fr']}},
+                                    ]
+                                }
+                            }
+                        }
+                    }
+                ]
+            }
+        ]
+        """
+        queries = [
+            kf_pair
+            for fd in self.filter_definitions.values()
+            for kf_pair in fd.get_query_fragment(data)
+        ]
+        nested_query = {
+            "bool": {
+                "must":
+                # queries => map(pluck("fragment")) => flatten()
+                [clause for kf_pair in queries for clause in kf_pair["fragment"]]
+            }
+        }
+        return (
+            [
+                {
+                    "key": self.name,
+                    "fragment": [
+                        {"nested": {"path": self.path, "query": nested_query}}
+                    ],
+                }
+            ]
+            if queries
+            else []
+        )
+
+    # pylint: disable=arguments-differ
+    def get_aggs_fragment(self, queries, data, *args, **kwargs):
```

Comment:
No docstring? Is this not a pylint error?

Reply:
maybe because of the pylint rule above? anyway, let me add a docstring...